sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnStats.scala

```diff
@@ -295,6 +295,28 @@ private[columnar] final class BinaryColumnStats extends ColumnStats {
     Array[Any](null, null, nullCount, count, sizeInBytes)
 }
 
+private[columnar] final class IntervalColumnStats extends ColumnStats {
```

Comment:
(This is not related to this pr though...) Can we already support column stats for intervals? Can you check?
https://github.com/apache/spark/blob/16f1b23d75c0b44aac61111bfb2ae9bb0f3fab68/sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala#L139
https://github.com/apache/spark/blob/16f1b23d75c0b44aac61111bfb2ae9bb0f3fab68/sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala#L238

Reply:
yes, these can be set to true for intervals

Reply:
Ah, I see. Thanks for the check. If you have time, do you wanna make a pr for that?

Reply:
yea, I'd like to make a followup after this pull request, thanks for notice me this.